sn_networking/src/event.rs

```diff
@@ -1067,6 +1066,9 @@ impl SwarmDriver {
                             "Early completion with the first copy of chunk {:?}",
                             chunk.name()
                         );
+
+                        // Stop tracking the query task by removing the entry and consume the sender.
+                        let (sender, ..) = entry.remove();
```

Comment:
can you make this `entry.remove()` call AFTER the `sender.send` ?
i.e. using the `sender` from the previous `entry.get_mut()` ?
same to the other place of using `entry`

Reply:
That's not possible, `send` consumes `self` (https://docs.rs/tokio/latest/tokio/sync/oneshot/struct.Sender.html#method.send).